In [1]:
import numpy as np

In [2]:
a = np.arange(6)
a2 = a[np.newaxis, :]
a2.shape

(1, 6)

In [3]:
a = np.array([1, 2, 3, 4, 5, 6])

In [4]:
a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])


In [5]:
print(a[0])

[1 2 3 4]


In [6]:
np.zeros(2)

array([0., 0.])

In [7]:
np.ones(2)

array([1., 1.])

In [8]:
np.empty(2)

array([1., 1.])

In [9]:
np.arange(4)

array([0, 1, 2, 3])

In [10]:
np.arange(2, 9, 2)

array([2, 4, 6, 8])

In [11]:
x = np.ones(2, dtype=np.int64)

In [12]:
x

array([1, 1])

In [13]:
arr = np.array([2, 1, 5, 3, 7, 4, 6, 8])

In [14]:
np.sort(arr)

array([1, 2, 3, 4, 5, 6, 7, 8])

In [15]:
a = np.array([1, 2, 3, 4])
b = np.array([5, 6, 7, 8])


In [17]:
np.concatenate((a, b))

array([1, 2, 3, 4, 5, 6, 7, 8])

In [18]:
x = np.array([[1, 2], [3, 4]])
y = np.array([[5, 6]])


In [19]:
np.concatenate((x, y), axis=0)

array([[1, 2],
       [3, 4],
       [5, 6]])

In [20]:
x.ndim

2

In [21]:
x.size

4

In [22]:
x.shape

(2, 2)

In [23]:
a = np.arange(6)

In [24]:
print(a)

[0 1 2 3 4 5]


In [25]:
b = a.reshape(3, 2)
print(b)

[[0 1]
 [2 3]
 [4 5]]


In [26]:
data = np.array([1, 2, 3])
data[1]



2

In [27]:
data[0:2]


array([1, 2])

In [28]:
data[1:]


array([2, 3])

In [29]:
data[-2:]

array([2, 3])

In [30]:
data = np.array([1, 2])
ones = np.ones(2, dtype=int)
data + ones


array([2, 3])

In [31]:
data - ones

array([0, 1])

In [32]:
data * data

array([1, 4])

In [33]:
data / data


array([1., 1.])

In [34]:
a = np.array([1, 2, 3, 4])

In [35]:
a.sum()

10

In [36]:
b = np.array([[1, 1], [2, 2]])

In [37]:
b

array([[1, 1],
       [2, 2]])

In [38]:
b.sum(axis=0)

array([3, 3])

In [39]:
b.sum(axis=1)

array([2, 4])

In [40]:
b*5

array([[ 5,  5],
       [10, 10]])

In [41]:
data.max()

2

In [42]:
data.min()

1

In [43]:
data.sum()

3

In [44]:
data = np.array([[1, 2], [3, 4], [5, 6]])
data

array([[1, 2],
       [3, 4],
       [5, 6]])

In [45]:
data[0, 1]

2

In [46]:
data[1:3]

array([[3, 4],
       [5, 6]])

In [47]:
a = np.array([11, 11, 12, 13, 14, 15, 16, 17, 12, 13, 11, 14, 18, 19, 20])

In [48]:
unique_values = np.unique(a)

In [49]:
print(unique_values)

[11 12 13 14 15 16 17 18 19 20]


In [50]:
arr = np.array([1, 2, 3, 4, 5, 6, 7, 8])

In [51]:
reversed_arr = np.flip(arr)

In [52]:
print('Reversed Array: ', reversed_arr)

Reversed Array:  [8 7 6 5 4 3 2 1]


In [53]:
# Importing pandas library
import pandas as pd

# Creating and initializing a nested list
age = [['Aman', 95.5, "Male"], ['Sunny', 65.7, "Female"],
       ['Monty', 85.1, "Male"], ['toni', 75.4, "Male"]]

# Creating a pandas dataframe
df = pd.DataFrame(age, columns=['Name', 'Marks', 'Gender'])

# Printing dataframe
df


Name  Marks  Gender
0   Aman   95.5    Male
1  Sunny   65.7  Female
2  Monty   85.1    Male
3   toni   75.4    Male

In [54]:
# Importing Numpy package
import numpy as np

# Creating a 3-D numpy array using np.array()
org_array = np.array([[23, 46, 85],[43, 56, 99],[11, 34, 55]])

# Printing the Numpy array
print(org_array)


[[23 46 85]
 [43 56 99]
 [11 34 55]]





## Pandas Dataframe  vs NumPy arrays/matrices


|Pandas Dataframe               |NumPy arrays/matrices                |
|-------------------------------|-----------------------------|
|When we have to work on **Tabular data**, we prefer the pandas module.            |When we have to work on **Numerical data**, we prefer the numpy module.            |
|The powerful tools of pandas are **Data frame and Series.**           |Whereas the powerful tool of _numpy_ is **Arrays.**           |
|Pandas consume **more**  **memory**.|_Numpy_ is **memory efficient.**|
|Pandas has a better performance when a number of rows is  **500K or more.** |Numpy has a better performance when number of rows is  **50K or less.**|
|Indexing of the  _pandas_  series is  **very slow**  as compared to  _numpy_  arrays.|Indexing of  numpy  Arrays is **very fast**.|
|Pandas  offer a have2d table object called  **DataFrame.**|Numpy is capable of providing  **multi-dimensional arrays.**|



In [56]:
data = [1, 4, 5, 7, 8, 8, 10]
#calculate average value of array
np.mean(data)


6.142857142857143

In [57]:

#calcualte average value of array
np.average(data)

6.142857142857143

In [58]:
np.average(data, weights=(.1, .2, .4, .05, .05, .1, .1))

5.449999999999999

Here is the formula that np.average() used to calculate this value:

Weighted Average = 1* .1 + 4* .2 + 5* .4 + 7* .05 + 8* .05 + 8* .1 + 10* .1 = 5.45.


### np.average takes an optional weight parameter. If it is not supplied it is same as np.mean